state-chain/pallets/cf-lending-pools/src/general_lending.rs

```diff
@@ -1229,6 +1383,24 @@ impl<T: Config> LendingApi for Pallet<T> {
 			Ok(())
 		})
 	}
+
+	fn set_voluntary_liquidation_flag(borrower_id: Self::AccountId, value: bool) -> DispatchResult {
+		LoanAccounts::<T>::try_mutate(&borrower_id, |maybe_account| {
+			let loan_account = maybe_account.as_mut().ok_or(Error::<T>::LoanAccountNotFound)?;
+
+			loan_account.voluntary_liquidation_requested = value;
+
+			// Re-evaluate liquidation status taking the new value of the flag into
+			// account. This may or may not update the actual status depending on LTV.
+			// Note that we ignore the error to allow the user to update
+			// the voluntary liquidation flag even if oracle prices aren't available.
+			if let Ok(ltv) = loan_account.derive_ltv() {
+				loan_account.update_liquidation_status(&borrower_id, ltv);
+			}
```

Comment:
Do we really need to do this? It will be checked again during the upkeep phase, no?

Reply:
The thinking was to react to the user's request as quickly as possible, but I suppose it doesn't make a big difference if the user has to wait for one block, so I will simplify.